provider/lib.go

```diff
@@ -17,3 +21,74 @@ func FilterFilePattern(regex string, filepath string) (bool, error) {
 	return filebool.Match([]byte(filepath)), nil
 
 }
+
+func FindFilesMatchingPattern(root, pattern string) ([]string, error) {
+	matches := []string{}
+	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		matched, err := FilterFilePattern(pattern, d.Name())
+		if err != nil {
+			return err
+		}
+		if matched {
+			matches = append(matches, path)
+		}
+		return nil
+	})
+	return matches, err
+}
+
+func GetFiles(filePaths []string, rootpath string) ([]string, error) {
+	var xmlFiles []string
```

Comment:
any references to XML files should be removed/renamed from this function, since this is a generic function for getting files of any kind.